src/filter/macros.rs

```diff
@@ -0,0 +1,139 @@
+#[macro_export]
+// macro to initialize vectors with a given capacity to reduce copy / paste
+macro_rules! store_vec {
+    // $name: the name of the variable
+    // $type: Type of data stored in vector (Mat5 / Vec5)
+    // $capacity: How much space to allocate
+    ($capacity:expr ; $($name:ident : $type:ty),+ ) => {
+        $(
+            let mut $name : Vec<$type> = Vec::with_capacity($capacity);
+        )+
+        // let mut $name: Vec<$type> = Vec::with_capacity($capacity);
+    };
+}
+
+
+#[macro_export]
+/// Push a value into an iterator. Saves repeating vec.push(asdad)
+/// Easier to read this way.
+macro_rules! push {
+    ($($item:expr => $location:ident),*) => {
+        $(
+            $location.push($item);
+        )*
+    };
+    (remove: $index:expr; $($pop_vec:ident => $dest_vec:ident),+) =>{
+        $(
+            let removed_val = $pop_vec.remove($index);
+            push!{removed_val => $dest_vec};
+        )+
+    };
+    (remove: $index:expr;$($pop_vec:ident),+ ) => {
+        $($pop_vec.remove($index);)+
+    };
+}
+
+#[macro_export]
+/// Get a reference to an item in a vector using .get_unchecked(_)
+macro_rules! get_unchecked {
+    ($index:expr; $($vector:ident => $destination:ident),+) => {
+        $(
+            let $destination = 
+                unsafe {
+                    $vector.get($index).expect("get_unchecked!{} fetched something out of bounds")
```

Comment:
Exactly.

Reply:
I was originally using `get_unchecked` but instead used `.get().expect()` to see where my indexing was incorrect in the for loop. Since `get_unchecked` has less overhead ill be sure to switch it back.

Reply:
Ok